### ISSUE TITLE
Release object_store `0.12.0` (API breaking) Around Feb 30 2025

### ISSUE THREAD

**Issue by alamb** [apache/arrow-rs#6903]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
Per @tustvold  on
- https://github.com/apache/arrow-rs/issues/6596

> and then do a breaking release late February in advance of the scheduled March breaking arrow release

Previous release:
- https://github.com/apache/arrow-rs/issues/6902

**Describe the solution you'd like**
Let's plan for a major release

- [ ] Update version + make release notes
- [ ] Prepare release candidate + begin vote thread
- [ ] Release to crates.io

**Items to potentially resolve**
- [x] https://github.com/apache/arrow-rs/pull/7183
- [x] https://github.com/apache/arrow-rs/pull/7181
- [x] https://github.com/apache/arrow-rs/pull/7170
- [x] https://github.com/apache/arrow-rs/pull/7171
- [x] https://github.com/apache/arrow-rs/issues/6818

**Describe alternatives you've considered**

**Additional context**
See discussion on https://github.com/apache/arrow-rs/issues/6596

**Comment by alamb** [comment]
I plan to assemble a release candidate for this release later in the week or early next week

**Comment by tustvold** [comment]
I think it would be good to try to get the following into this release if possible:

* https://github.com/apache/arrow-rs/pull/7183
* https://github.com/apache/arrow-rs/pull/7181
* https://github.com/apache/arrow-rs/pull/7170 (if we go down that path)

**Comment by Xuanwo** [comment]
Hi, I personally hope we can hold off on https://github.com/apache/arrow-rs/pull/7170 until we resolve https://github.com/apache/arrow-rs/issues/7171. 

Alternatively, we could implement this feature in `http::Extension`, as mentioned by @tustvold, which wouldn't require changing the `object_store` API. This way, the changes would be scoped only to DF.

**Comment by alamb** [comment]
Thanks @tustvold and @Xuanwo  -- I have added those items to the list on this ticket

**Comment by tustvold** [comment]
I'm going to get WASM32 over the line and then I think we should be good to go

**Comment by alamb** [comment]
Awesome -- @tustvold  do you want to run this release or shall I plan to ?

**Comment by alamb** [comment]
> I'm going to get WASM32 over the line and then I think we should be good to go

I added this ticket to the list. Is that what you mean?
- https://github.com/apache/arrow-rs/issues/7227

**Comment by Xuanwo** [comment]
`Feb 30 2025` never reaches 😆

**Comment by alamb** [comment]
> `Feb 30 2025` never reaches 😆

Yeah, perhaps it was inevitable that this release would be slightly delayed 😆

**Comment by tustvold** [comment]
> I added this ticket to the list. Is that what you mean?

That's actually a follow up that can be done in a later release, the initial support ticket is #6818.

**Comment by alamb** [comment]
> That's actually a follow up that can be done in a later release, the initial support ticket is [#6818](https://github.com/apache/arrow-rs/issues/6818).

Updated


> > do you want to run this release or shall I plan to ?
> 
> Up to you, if you're happy to that'd be amazing from my perspective but I can also 😅

If you could do it that would be great. I will be happy to help test / do other legwork required?

**Comment by tustvold** [comment]
Ok I'll try to find some time either this evening or tomorrow

**Comment by tustvold** [comment]
I tried to cut this release, but have issues as my Apache blessed GPG keys have expired. I'm afraid I don't have any more time to dedicate to this at this time, but can potentially try to find some more time this weekend.

**Comment by alamb** [comment]
I'll try and make an RC as @tustvold  did most of the work

**Comment by alamb** [comment]
Thank you for testing @criccomini 🙏

**Comment by alamb** [comment]
The release is approved and available on crates.io
- mailing list: https://lists.apache.org/thread/483b63g6djkbmgft2fqkoyhd45612f3q
- https://crates.io/crates/object_store/0.12.0

The next incremental release is tracked with 
- https://github.com/apache/arrow-rs/issues/7085

Thanks everyone!

**Comment by tustvold** [comment]
Thank you for doing this Andrew

**Comment by alamb** [comment]
`label_issue.py` automatically added labels {'object-store'} from #7234

**Comment by alamb** [comment]
`label_issue.py` automatically added labels {'documentation'} from #7086

**Comment by alamb** [comment]
`label_issue.py` automatically added labels {'parquet'} from #7045

**Comment by alamb** [comment]
`label_issue.py` automatically added labels {'arrow'} from #7045

**Comment by alamb** [comment]
`label_issue.py` automatically added labels {'next-major-release'} from #6619

**Comment by alamb** [comment]
Migrating from arrow-rs issue [#6903](https://github.com/apache/arrow-rs/issues/6903)
